framework/Models.py

```diff
@@ -1468,59 +1451,81 @@ def createNewInput(self,currentInput,samplerType,**Kwargs):
+        @ Out, returnValue, dict, This holds the output information of the evaluated sample.
     """
+    inputFiles = self.createNewInput(myInput, samplerType, **kwargs)
     self.currentInputFiles, metaData = (copy.deepcopy(inputFiles[0]),inputFiles[1]) if type(inputFiles).__name__ == 'tuple' else (inputFiles, None)
     returnedCommand = self.code.genCommand(self.currentInputFiles,self.executable, flags=self.clargs, fileArgs=self.fargs, preExec=self.preExec)
-    if type(returnedCommand).__name__ != 'tuple':
-      self.raiseAnError(IOError, "the generateCommand method in code interface must return a tuple")
-    if type(returnedCommand[0]).__name__ != 'list':
-      self.raiseAnError(IOError, "the first entry in tuple returned by generateCommand method needs to be a list of tuples!")
+
+    ## Given that createNewInput can only return a tuple, I don't think these
+    ## checks are necessary (keeping commented out until someone else can verify):
+    # if type(returnedCommand).__name__ != 'tuple':
+    #   self.raiseAnError(IOError, "the generateCommand method in code interface must return a tuple")
+    # if type(returnedCommand[0]).__name__ != 'list':
+    #   self.raiseAnError(IOError, "the first entry in tuple returned by generateCommand method needs to be a list of tuples!")
     executeCommand, self.outFileRoot = returnedCommand
-    uniqueHandler = inputFiles[1]['uniqueHandler'] if 'uniqueHandler' in inputFiles[1].keys() else 'any'
-    if 'prefix' in inputFiles[1].keys():
-      identifier    = inputFiles[1]['prefix']
-    else:
-      identifier    = None
-      inputFiles[1]['prefix'] = 'None'
-    jobHandler.addExternal(executeCommand,self.outFileRoot,metaData.pop('subDirectory'),identifier=identifier,metadata=metaData,codePointer=self.code,uniqueHandler = uniqueHandler)
+
+    precommand = kwargs['precommand']
+    mpiCommand = kwargs['mpiCommand']
+    threadingCommand = kwargs['threadingCommand']
+    postcommand = kwargs['postcommand']
+    bufferSize = kwargs['bufferSize']
+    fileExtensionsToDelete = kwargs['deleteOutExtension']
+    deleteSuccessfulLogFiles = kwargs['delSucLogFiles']
+
+    codeLogFile = self.outFileRoot
+    if codeLogFile is None:
+      codeLogFile = os.path.join(metaData['subDirectory'],'generalOut')
+
+    ## This could cause issues as the temporary changing of the directory on
+    ## this thread could affect the main thread's working directory, I would
+    ## prefer to copy the environment and then update the working directory of
+    ## the new environment, but I am not sure if this is OS agnostic? --DPM 4/25/17
+    ## push directory
+    # oldDir = os.getcwd()
+    # os.chdir(metaData['subDirectory'])
+
+    ## This is the preferred alternative, but we should ensure that it works
+    ## on all systems
+    sampleDirectory = os.path.join(os.getcwd(),metaData['subDirectory'])
+    localenv = dict(os.environ)
+    localenv['PWD'] = str(sampleDirectory)
```

Comment:
Well, the problem is if we don't set PWD in the copied environment it will actually point to the wrong directory, so that must be specified.  I am testing what happens if we omit the cwd parameter from the call.